### PR TITLE
fix(IEC): move server into services/iec

### PR DESCRIPTION
**Pull request by 346010762**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
   move server into services dir, fix lint warnings
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/iec/ TESTARGS='-run TestAccIecServerResource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iec/ -v -run TestAccIecServerResource_basic -timeout 360m -parallel 4
=== RUN   TestAccIecServerResource_basic
--- PASS: TestAccIecServerResource_basic (140.13s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec       140.174s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_server_test.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  cidr = "192.168.0.0/16"
+  mode = "CUSTOMER"
+}
+
+resource "huaweicloud_iec_vpc_subnet" "subnet_test" {
+  name = "subnet-%s"
+  cidr = "192.168.0.0/16"
+  gateway_ip = "192.168.0.1"
+  vpc_id  = huaweicloud_iec_vpc.vpc_test.id
+  site_id = data.huaweicloud_iec_sites.sites_test.sites[0].id
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_server_test.go]: align the equal signs

File: huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_server_test.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  description = "this is a test group"
+}
+
+resource "huaweicloud_iec_security_group_rule" "rule_test" {
+  direction      = "ingress"
+  port_range_min = 445
+  port_range_max = 445
+  protocol       = "tcp"
+  security_group_id = huaweicloud_iec_security_group.secgroup_test.id
+  remote_ip_prefix = "0.0.0.0/0"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_server_test.go]: ditto

File: huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_server_test.go (hunk 1 above)
Comment (346010762) [reply huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_server_test.go]: done

File: huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_server_test.go (hunk 2 above)
Comment (346010762) [reply huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_server_test.go]: done

File: huaweicloud/services/iec/resource_huaweicloud_iec_server.go [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack/ecs/v1/cloudservers"
+	"github.com/chnsz/golangsdk/openstack/iec/v1/cloudvolumes"
+	"github.com/chnsz/golangsdk/openstack/iec/v1/common"
```
Comment (ShiChangkuo) [inline huaweicloud/services/iec/resource_huaweicloud_iec_server.go]: it's better to use `ieccommon` as the alias of iev/v1/common, and keep `huaweicloud/common` as it is.

Comment (346010762) [reply huaweicloud/services/iec/resource_huaweicloud_iec_server.go]: done

**Pull request merged**